xsl/assembly/effectivity.xsl

```diff
@@ -21,82 +24,192 @@
   <xsl:param name="effectivity.userlevel" />
   <xsl:param name="effectivity.vendor" />
   <xsl:param name="effectivity.wordsize" />
+  <xsl:param name="effectivity.separator" />
+
+  <xsl:param name="profile.arch" select="$effectivity.arch"/>
+  <xsl:param name="profile.audience" select="$effectivity.audience"/>
+  <xsl:param name="profile.condition" select="$effectivity.condition" />
+  <xsl:param name="profile.conformance" select="$effectivity.conformance"/>
+  <xsl:param name="profile.os" select="$effectivity.os"/>
+  <xsl:param name="profile.outputformat" select="$effectivity.outputformat"/>
+  <xsl:param name="profile.revision" select="$effectivity.revision"/>
+  <xsl:param name="profile.security" select="$effectivity.security"/>
+  <xsl:param name="profile.userlevel" select="$effectivity.userlevel"/>
+  <xsl:param name="profile.vendor" select="$effectivity.vendor"/>
+  <xsl:param name="profile.wordsize" select="$effectivity.wordsize"/>
+  <xsl:param name="profile.lang"/>
+  <xsl:param name="profile.revisionflag"/>
+  <xsl:param name="profile.role"/>
+  <xsl:param name="profile.status"/>
+  <xsl:param name="profile.attribute"/>
+  <xsl:param name="profile.value"/>
+  <xsl:param name="profile.separator" select="$effectivity.separator" />
+
+
 
   <xsl:template match="d:module[@resourceref] | d:structure[@resourceref]" mode="evaluate.effectivity">
     <xsl:apply-templates mode="evaluate.effectivity" />
   </xsl:template>
   
   <xsl:template match="d:filterout" mode="evaluate.effectivity">
-    <xsl:variable name="effectivity.attribute.arch" select="@arch" />
-    <xsl:variable name="effectivity.attribute.audience" select="@audience" />
-    <xsl:variable name="effectivity.attribute.condition" select="@condition" />
-    <xsl:variable name="effectivity.attribute.conformance" select="@conformance" />
-    <xsl:variable name="effectivity.attribute.os" select="@os" />
-    <xsl:variable name="effectivity.attribute.outputformat" select="@outputformat" />
-    <xsl:variable name="effectivity.attribute.revision" select="@revision" />
-    <xsl:variable name="effectivity.attribute.security" select="@security" />
-    <xsl:variable name="effectivity.attribute.userlevel" select="@userlevel" />
-    <xsl:variable name="effectivity.attribute.vendor" select="@vendor" />
-    <xsl:variable name="effectivity.attribute.wordsize" select="@wordsize" />
+    
+    <xsl:variable name="effectivity.match.arch">
+      <!-- <xsl:message>INFO: The value of effectivity.arch as passed in is <xsl:value-of select="$effectivity.arch" /> ... </xsl:message>
+      <xsl:message>INFO: The value of @arch as read from the XML file is <xsl:value-of select="@arch" /> ... </xsl:message> -->
+      <xsl:call-template name="cross.compare">
```

Comment:
My processor goes into what seems to be an infinite loop here, if I set `effectivity.arch`. Let's take a look at my test setup?